### DISCUSSION TITLE
Can I send a big size via RabbitMQ Stream?

### DISCUSSION THREAD

**Discussion by chienhust** [rabbitmq/rabbitmq-server#8134] [Questions]
I'm learning about RabbitMQ Stream, I see that i can set maximum stream size to 20 GB by below argument:
`arguments.put("x-max-length-bytes", 20_000_000_000); // maximum stream size: 20 GB`
(refer to https://www.rabbitmq.com/streams.html#declaring)
So, can I send a big file like a docker image (about 5GB) or a zip file (about 1GB) via RabbitMQ Stream? and how to do that?
Thank you very much.

**Comment by kjnilsson** [comment]
RabbitMQ streams are designed for large amounts of data but individual messages of those sizes I could not recommend that you try.

it would be much better to e.g. set up a shared docker repository and publish your docker image there instead then send a message through rabbitmq notifying interested parties that there is a new version of the image.

**Comment by michaelklishin** [comment]
Storing large blobs on a blob store (or a more specialized store such as a Docker image registry) and passing URIs or IDs around is the recommended way, for quorum queues, non-mirrored classic queues, and streams.
